### ISSUE TITLE
Spawning in GMS regardless of settings

### ISSUE THREAD

**Issue by GenerationMC** [Multiverse/Multiverse-Core#2299]
I keep spawning in gms on my server even though I left in gmc and even in worlds that game mode is set to gmc such as my creative world. It switches back the gms every time regardless. 

My server properties are set to false on force gamemode so it's not that.

` Creative:
    ==: MVWorld
    hidden: 'false'
    alias: Creative
    color: WHITE
    style: NORMAL
    pvp: 'false'
    scale: '1.0'
    respawnWorld: ''
    allowWeather: 'false'
    difficulty: EASY
    spawning:
      ==: MVSpawnSettings
      animals:
        ==: MVSpawnSubSettings
        spawn: 'false'
        spawnrate: '-1'
        exceptions: []
      monsters:
        ==: MVSpawnSubSettings
        spawn: 'false'
        spawnrate: '-1'
        exceptions: []
    entryfee:
      ==: MVEntryFee
      amount: '0.0'
    hunger: 'false'
    autoHeal: 'true'
    adjustSpawn: 'true'
    portalForm: ALL
    gameMode: CREATIVE
    keepSpawnInMemory: 'true'
    spawnLocation:
      ==: MVSpawnLocation
      x: -82.5
      y: 4.0
      z: 66.5
      pitch: 12.003322
      yaw: 3.6253111
    autoLoad: 'true'
    bedRespawn: 'false'
    worldBlacklist: []
    environment: NORMAL
    seed: '6346887350640753594'
    generator: 'null'
    playerLimit: '-1'
    allowFlight: 'true'`

**Comment by benwoo1110** [comment]
What's your server version and type?

Also can you reproduce the issue with only mv installed?

Did you happen to also give `mv.bypass.gamemode.GAMEMODE` permission node?

**Comment by GenerationMC** [comment]
PaperMC 1.15.2 & Multiverse-Core-4.1.0

Have taken all plugins off and just have this right now and seems to have fixed the gms but I noticed another issue that I cannot seem to fix no matter what I try. It refuses to respawn in the correct world after death. 

: No essentials spawn to conflict or any other plugins
: Have done /mv setspawn for each world
: Have set respawn world for each world in worlds.yml

Regardless it keeps spawning back on the world where my bed is. We do not want to turn bed respawning off because it's needed in a couple of our survival worlds. We just want the spawns to do what they are suppose to do and actually act as that worlds spawn or respawn point. Any idea why it refuses to work?

**Comment by benwoo1110** [comment]
whoops sleepy me accidentally closed the issue.

you can try turn off bed respawn for the worlds you don't want to have it with `/mvm set bedRespawn false <worldname>`

for respawnWorld, try using the commands instead: `/mvm set respawnWorld <respawn_world> <death_world>`

Also, do you have mv-inv installed?

**Comment by GenerationMC** [comment]
It went to the world with my bed in regardless of the bed respawn settings of that particular world. I decided to just set it to false in all worlds now because been trying to fix the issue for hours now with no luck so just gave in. Its respawning correctly now.

The gamemode issue was due to my hub plugin. I didn't notice they had a default player join event set for game mode so thats all sorted now too.

Thanks for your help.
